tools/cldr-code/src/main/java/org/unicode/cldr/util/NameType.java

```diff
@@ -0,0 +1,275 @@
+package org.unicode.cldr.util;
+
+import java.util.Locale;
+
+public enum NameType {
+    NONE,
+    LANGUAGE,
+    SCRIPT,
+    TERRITORY,
+    VARIANT,
+    CURRENCY,
+    CURRENCY_SYMBOL,
+    TZ_EXEMPLAR,
+    TZ_GENERIC_LONG,
+    TZ_GENERIC_SHORT,
+    TZ_STANDARD_LONG,
+    TZ_STANDARD_SHORT,
+    TZ_DAYLIGHT_LONG,
+    TZ_DAYLIGHT_SHORT,
+    KEY,
+    KEY_TYPE,
+    SUBDIVISION;
+
+    // The order of rows must correspond to INDEX_LANGUAGE, INDEX_SCRIPT, etc.
+    // Caution: the presence of "key|type" presents complications for refactoring.
+    // The row with "key|type" has four strings, while the others have three.
+    private static final String[][] NameTable = {
+        {"//ldml/localeDisplayNames/languages/language[@type=\"", "\"]", "language"},
+        {"//ldml/localeDisplayNames/scripts/script[@type=\"", "\"]", "script"},
```

Comment:
NameTable is moved from CLDRFile, otherwise unchanged. Further refactoring could be beneficial, to avoid the entanglement with integers INDEX_LANGUAGE, etc., but at least that entanglement is now encapsulated inside this class

Reply:
yeah, actually something that works with XPathParts  (thus cache) - an XPathCalculator (builder) or something could be a completely different utility.  That's what this is, it knows how to make an XPath for certain purposes.